certification/internal/policy/container/has_unique_tag.go

```diff
@@ -25,17 +25,34 @@ type hasUniqueTagCheck struct {
 
 func (p *hasUniqueTagCheck) Validate(ctx context.Context, imgRef certification.ImageReference) (bool, error) {
 	imgRepo := fmt.Sprintf("%s/%s", imgRef.ImageRegistry, imgRef.ImageRepository)
-	tags, err := p.getDataToValidate(ctx, imgRepo)
-	if err != nil {
-		return false, fmt.Errorf("Error: failed to get tags list for %s: %v", imgRepo, err)
+
+	tags := make([]string, 0)
+	var err error
+	// if sha is passed in `/tags/list` must be exposed and available to validate that the image is being tagged properly
+	if strings.HasPrefix(imgRef.ImageTagOrSha, "sha256:") {
+		tags, err = p.getDataToValidate(ctx, imgRepo)
+		if err != nil {
+			return false, fmt.Errorf("failed to get tags list for %s when sha256 provided: %v", imgRepo, err)
```

Comment:
It looks like the only difference between these is just the log message. I don't think that necessarily adds any value. This can be reduced to a single `if this || that`.

Reply:
Along this line, to improve readability, I'd say that these two cases should be methods or functions. Then it literally becomes `if this || that`.

Reply:
This has been simplified, into one `if` statement.